lib/api/src/lib/stories.ts

```diff
@@ -180,120 +174,162 @@ export const denormalizeStoryParameters = ({
   }));
 };
 
-const STORY_KIND_PATH_SEPARATOR = /\s*\/\s*/;
+const TITLE_PATH_SEPARATOR = /\s*\/\s*/;
 
-export const transformStoryIndexToStoriesHash = (
-  index: StoryIndex,
+// We used to received a bit more data over the channel on the SET_STORIES event, including
+// the full parameters for each story.
+type PreparedStoryIndexEntry = StoryIndexEntry & {
+  parameters?: Parameters;
+  argTypes?: ArgTypes;
+  args?: Args;
+  initialArgs?: Args;
+};
+export interface PreparedStoryIndex {
+  v: number;
+  entries: Record<StoryId, PreparedStoryIndexEntry>;
+}
+
+export const transformSetStoriesStoryDataToStoriesHash = (
+  data: SetStoriesStoryData,
   { provider }: { provider: Provider }
-): StoriesHash => {
-  const countByTitle = countBy(Object.values(index.entries), 'title');
-  const input = Object.entries(index.entries).reduce((acc, [id, { title, name, importPath }]) => {
-    const docsOnly = name === 'Page' && countByTitle[title] === 1;
-    acc[id] = {
-      id,
-      kind: title,
-      name,
-      parameters: { fileName: importPath, options: {}, docsOnly },
-    };
-    return acc;
-  }, {} as StoriesRaw);
+) =>
+  transformStoryIndexToStoriesHash(transformSetStoriesStoryDataToPreparedStoryIndex(data), {
+    provider,
+  });
+
+const transformSetStoriesStoryDataToPreparedStoryIndex = (
+  stories: SetStoriesStoryData
+): PreparedStoryIndex => {
+  const entries: PreparedStoryIndex['entries'] = Object.entries(stories).reduce(
+    (acc, [id, story]) => {
+      if (!story) return acc;
+
+      const { docsOnly, fileName, ...parameters } = story.parameters;
+      const base = {
+        title: story.kind,
+        id,
+        name: story.name,
+        importPath: fileName,
+      };
+      if (docsOnly) {
+        acc[id] = {
+          type: 'docs',
+          ...base,
+        };
+      } else {
+        const { argTypes, args, initialArgs } = story;
+        acc[id] = {
+          type: 'story',
+          ...base,
+          parameters,
+          argTypes,
+          args,
+          initialArgs,
+        };
+      }
+      return acc;
+    },
+    {} as PreparedStoryIndex['entries']
+  );
 
-  return transformStoriesRawToStoriesHash(input, { provider, prepared: false });
+  return { v: 4, entries };
```

Comment:
Does this `v: 4` line up with `index.json`'s `v: 4`?

Reply:
Yes

Reply:
Would be good to use the same types eventually